i18n

tcarreira opened the pull request:
fix #21 
- Support for i18n
- Portuguese (pt_PT) translation
- exampleSite with one translation
- show only non-empty taxonomies (so translated taxonomies don't show up)

tcarreira commented:
You may see translations in
- top menu 
- every translated article

If you open http://localhost:1313/post/markdown-syntax/ (the only translated article) you will see the reference for other translations. This is not super pretty, but the main idea is there.
Let me know if it can be improved.

tcarreira commented:
Also, the last commit is not really part of this feature (but it is almost required for it).
If for some reason you reject this PR, cherry-pick that commit :ok_hand:

razonyang commented:
LGTM. But it would be better to support the following features:

- Multilingual menu. Currently, the menu is hard code in `data/menu.yaml`. I think it is better to define menus in config file(https://gohugo.io/content-management/multilingual/#menus).
- Sidebar taxonomy links are broken in other languages.

tcarreira commented:
Agreed. Thanks for the feedback (it's my first time using Hugo). 
I'll try to fix those things this week.
Do you think it is better to merge this as it is and fix it later (as it does not break any current feature)? Or delay the merge with everything working?

razonyang commented:
I am not a maintainer of this repo and I have not right to merge this PR :joy:. ping @onweru.

onweru commented:
@tcarreira,

> Do you think it is better to merge this as it is and fix it later (as it does not break any current feature)? Or delay the merge with everything working?
 
The latter; let's get it right first.

tcarreira commented:
Hey @onweru I've made it.

It was far difficult than I expected, I had to change how the menu is created. 
I've updated the README file to reflect that, please double check it.

I also added the i18n section to the README. I think it is OK.

One thing to note:
I don't like how the available languages menu is right now. It's not VERY bad, but I think a UX or frontend expert could do much better. Anyway, for the sake of exampleSite, I think it is OK.

Let me know what you think @razonyang

tcarreira commented:
Explicit note: this PR will break a website that only upgrades the theme (menus must be built differently). 
This should be part of a 2.0 version then.

razonyang commented:
Thanks for your huge work. This PR almost meet my needs for I18N, but the taxonomies links are weird for me. 
Currently, the taxonomies links(URL) contain the translated text. Such as `series` links: `http://localhost:1313/pt/s%C3%A9ries/`(`pt`)，`http://localhost:1313/zh/%E7%B3%BB%E5%88%97/`(`zh`). 
It looks ugly for me. I think it is better to use uniform link form like `/{lang}/categories`, `/{lang}/tags` and  `/{lang}/series`.
What do you think about this? @tcarreira @onweru 

PS: I had made a PR for this yesterday(tcarreira/hugo-clarity#1).

tcarreira commented:
It looks cleaner (and easier) to me.
I'll handle that PR (I've missed it, thank for the remind) and merge it all

tcarreira commented:
@razonyang I've merged your PR (kind of, it has diverged from the code I've made after that)
I believe it is well enough. Does that fill all requirements for i18n?

razonyang commented:
@tcarreira yes, it is good enough for me.

tcarreira commented:
@onweru it's ready for merging 👍

onweru commented:
@tcarreira & @razonyang for your work on this PR; your contribution is most appreciated. Merging.

tcarreira commented:
solves #22

onweru commented on assets/js/functions.js:
```diff
@@ -2,7 +2,7 @@
 const inline = ":inline";
 // variables read from your hugo configuration
-const parentURL = '{{ absURL "" }}';
+const parentURL = '{{ absLangURL "" }}';
```
Bug in #51 introduced in this PR /commit 😏, totally missed it.

tcarreira replied:
Oops, didn't realize it either :+1:

onweru commented:
@tcarreira & @razonyang, if you may, please chime in on #57. An additional PR might be needed to properly document this. Thanks in advance.

tcarreira commented:
commented there https://github.com/chipzoller/hugo-clarity/issues/57#issuecomment-702298311
